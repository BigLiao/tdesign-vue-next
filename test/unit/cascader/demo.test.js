import { mount } from '@vue/test-utils';
import base from '@/examples/cascader/demos/base.vue';

// unit test for component in examples.
describe('Cascader', () => {
  it('base demo works fine', () => {
    const wrapper = mount(base);
    expect(wrapper.element).toMatchSnapshot();
  });
});
