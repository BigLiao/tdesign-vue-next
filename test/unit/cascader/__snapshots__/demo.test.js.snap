// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cascader base demo works fine 1`] = `
<div>
  <div
    class="t-demo-cascader-title"
  >
    非受控用法
  </div>
  <div
    class="t-demo-cascader"
    modelvalue=""
  >
    <div
      class="t-popup-reference"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-cascader-dropdown"
          >
            
            <div
              class="t-cascader-panel t-cascader--normal"
            >
              <ul
                class="t-cascader-menu"
              >
                <li
                  class="t-cascader-item t-cascader-item__is-empty"
                >
                  暂无数据
                </li>
              </ul>
            </div>
            
            <!---->
          </div>
        </div>
      </transition-stub>
      
      <div
        class="t-cascader t-size-s t-demo-cascader t-demo-cascader"
        modelvalue=""
      >
        <span
          class="t-cascader-placeholder"
        >
          请选择
        </span>
        <svg
          class="t-fake-arrow t-cascader-icon"
          disabled="false"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </div>
      
    </div>
  </div>
  <div
    class="t-demo-cascader-title"
  >
    受控用法
  </div>
  <div
    class="t-demo-cascader"
    modelvalue=""
  >
    <div
      class="t-popup-reference"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-cascader-dropdown"
          >
            
            <div
              class="t-cascader-panel t-cascader--normal"
            >
              <ul
                class="t-cascader-menu"
              >
                <li
                  class="t-cascader-item t-cascader-item__is-empty"
                >
                  暂无数据
                </li>
              </ul>
            </div>
            
            <!---->
          </div>
        </div>
      </transition-stub>
      
      <div
        class="t-cascader t-size-m t-demo-cascader t-demo-cascader"
        modelvalue=""
      >
        <span
          class="t-cascader-placeholder"
        >
          请选择
        </span>
        <svg
          class="t-fake-arrow t-cascader-icon"
          disabled="false"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </div>
      
    </div>
  </div>
  <div
    class="t-demo-cascader-title"
  >
    受控+语法糖用法
  </div>
  <div
    class="t-demo-cascader"
    modelvalue="1.1"
  >
    <div
      class="t-popup-reference"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-cascader-dropdown"
          >
            
            <div
              class="t-cascader-panel t-cascader--normal"
            >
              <ul
                class="t-cascader-menu"
              >
                <li
                  class="t-cascader-item t-cascader-item__is-empty"
                >
                  暂无数据
                </li>
              </ul>
            </div>
            
            <!---->
          </div>
        </div>
      </transition-stub>
      
      <div
        class="t-cascader t-size-l t-demo-cascader t-demo-cascader"
        modelvalue="1.1"
      >
        <span
          class="t-cascader-placeholder"
        >
          请选择
        </span>
        <svg
          class="t-fake-arrow t-cascader-icon"
          disabled="false"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 5.7998L7.99274 10.0425L12.2361 5.79921"
            stroke="black"
            stroke-opacity="0.9"
            stroke-width="1.3"
          />
        </svg>
      </div>
      
    </div>
  </div>
</div>
`;
